### PR TITLE
#160451548 Setup an empty react project

### PR DESCRIPTION
**Pull request by e0works**
#### What does this PR do?
 - setup empty react project

#### Description of Task to be completed?
- add webpack configuration
- add webpack and react dependencies
- setup a dummy react router

#### How should this be manually tested?
- start the app by running ```yarn start:dev```
the app should open on port ```8080``` on your browser

#### What are the relevant pivotal tracker stories?
[#160451548](https://www.pivotaltracker.com/story/show/160451548, "create empty react app")

#### Screenshots (if appropriate)
screen shot 2018-10-29 at 4 24 03 pm

### CODE REVIEW

**Review by koiic — APPROVED** [review]
Looking good to me

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
```
Comment (KaiserPhemi) [inline .gitignore]: #### Critical
Please create a more comprehensive `.gitignore` file. Check [here](https://www.npmjs.com/package/gitignore)

**NB**
- Please install `gitignore` as a global dependency on your system. It shouldn't be part of the list of dependencies for the project

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="utf-8" />
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <title>Author's Haven</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" type="text/css" media="screen" href="main.css" />
```
Comment (KaiserPhemi) [inline public/index.html]: #### Critical
- Remove this link tag. We won't be needing it. `webpack` should add one if setup properly.

File: src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (KaiserPhemi) [inline src/components/Header.js:2]: #### Nitpick
Always comment your imports as done in the component above

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

File: src/components/Header.js (hunk 3 above)
Comment (e0works) [reply src/components/Header.js:2]: I have commented on all the imports

**Comment by e0works** [comment]
@KaiserPhemi, I have implemented the feedback.

**Review by Mrshinezee — APPROVED** [review]
LGTM

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**